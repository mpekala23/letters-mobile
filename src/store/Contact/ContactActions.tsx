import {
  Contact,
  SET_ADDING,
  SET_EXISTING,
  CLEAR_CONTACTS,
  ContactActionTypes,
} from "./ContactTypes";

export function setAdding(contact: Contact): ContactActionTypes {
  return {
    type: SET_ADDING,
    payload: contact,
  };
}

export function setExisting(contacts: Contact[]): ContactActionTypes {
  return {
    type: SET_EXISTING,
    payload: contacts,
  };
}

export function clearContacts() {
  return {
    type: CLEAR_CONTACTS,
    payload: null,
  };
}
