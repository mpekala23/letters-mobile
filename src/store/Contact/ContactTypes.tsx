import { Facility } from "types";

export const SET_ADDING = "contact/set_adding";
export const SET_EXISTING = "contact/set_existing";
export const CLEAR_CONTACTS = "contact/clear_contacts";

export interface Contact {
  id: number;
  state: string;
  firstName: string;
  lastName: string;
  inmateNumber: string;
  relationship: string;
  facility: Facility | null;
}

export interface ContactState {
  adding: Contact;
  existing: Contact[];
}

interface SetAddingAction {
  type: typeof SET_ADDING;
  payload: Contact;
}

interface SetExistingAction {
  type: typeof SET_EXISTING;
  payload: Contact[];
}

interface ClearContactsAction {
  type: typeof CLEAR_CONTACTS;
  payload: null;
}

export type ContactActionTypes =
  | SetAddingAction
  | SetExistingAction
  | ClearContactsAction;
