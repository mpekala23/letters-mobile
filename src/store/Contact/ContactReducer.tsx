import { ContactState, ContactActionTypes, SET_ADDING, SET_EXISTING } from './ContactTypes';

const initialState: ContactState = {
  adding: {
    state: '',
    first_name: '',
    last_name: '',
    inmate_number: '',
    relationship: '',
    facility: null,
  },
  existing: [],
};

export default function ContactReducer(
  state = initialState,
  action: ContactActionTypes
): ContactState {
  const currentState = { ...state };
  switch (action.type) {
    case SET_ADDING:
      currentState.adding = action.payload;
      return currentState;
    case SET_EXISTING:
      currentState.existing = action.payload;
      return currentState;
    default:
      return state;
  }
}
