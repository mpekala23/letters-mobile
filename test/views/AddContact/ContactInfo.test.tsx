import * as React from 'react';
import { ContactInfoScreen } from '@views';
import { render, toJSON, fireEvent } from '@testing-library/react-native';
import { SET_ADDING } from '@store/Contact/ContactTypes';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

const setup = (
  contactOverrides = {},
  userOverrides = {},
  routeOverrides = {}
) => {
  const navigation = {
    navigate: jest.fn(),
    addListener: jest.fn(),
    setParams: jest.fn(),
  };
  const route = {
    params: {
      addFromSelector: false,
      ...routeOverrides,
    },
  };
  const contact = {
    firstName: '',
    lastName: '',
    inmateNumber: '',
    relationship: '',
    facility: null,
    ...contactOverrides,
  };
  const user = {
    id: '6',
    firstName: 'Team',
    lastName: 'Ameelio',
    email: 'team@ameelio.org',
    cell: '4324324432',
    address1: 'Somewhere',
    zipcode: '12345',
    city: 'New Haven',
    state: 'CT',
    ...userOverrides,
  };
  const authInfo = {
    isLoadingToken: false,
    isLoggedIn: true,
  };
  const initialState = {
    adding: contact,
    existing: [],
  };
  const store = mockStore({
    contact: initialState,
    user: {
      authInfo,
      user,
    },
  });

  const StoreProvider = ({ children }: { children: JSX.Element }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return {
    navigation,
    store,
    ...render(
      <ContactInfoScreen
        navigation={navigation}
        contactState={initialState}
        route={route}
      />,
      {
        wrapper: StoreProvider,
      }
    ),
  };
};

describe('Contact Info Screen', () => {
  it('should match snapshot', () => {
    const { container } = setup(<ContactInfoScreen />);
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should load initial values for fields from the redux store', () => {
    const { getByPlaceholderText } = setup({
      firstName: 'First',
      lastName: 'Last',
      inmateNumber: '6',
      relationship: 'Sister',
    });
    expect(getByPlaceholderText('First Name').props.value).toBe('First');
    expect(getByPlaceholderText('Last Name').props.value).toBe('Last');
    expect(getByPlaceholderText('Inmate Number').props.value).toBe('6');
    expect(getByPlaceholderText('Relationship to Inmate').props.value).toBe(
      'Sister'
    );
  });

  it('should update the state databases to search when user inputs a valid state', () => {
    const { queryAllByText, getAllByText, getByPlaceholderText } = setup();
    fireEvent.changeText(getByPlaceholderText('State'), 'Iowa');
    expect(getAllByText('Iowa')).toBeDefined();
    const stateInput = getByPlaceholderText('State');
    fireEvent.changeText(stateInput, 'Not a valid state');
    fireEvent.changeText(stateInput, 'Kansas');
    expect(queryAllByText('Iowa').length).toBeFalsy();
    expect(getAllByText('Kansas')).toBeDefined();
  });
});
