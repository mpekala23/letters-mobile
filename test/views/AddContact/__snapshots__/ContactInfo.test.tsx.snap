// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Info Screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={1}
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <ScrollView
          keyboardShouldPersistTaps="handled"
          scrollEnabled={false}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "height": 20,
                "width": "100%",
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "white",
                "flex": 1,
                "paddingHorizontal": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 24,
                    }
                  }
                >
                  Add Contact
                </Text>
              </View>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={19}
                bbWidth={24}
                fill="none"
                focusable={false}
                height={19}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "margin": 16,
                    },
                    Object {
                      "flex": 0,
                      "height": 19,
                      "width": 24,
                    },
                  ]
                }
                vbHeight={19}
                vbWidth={24}
                width={24}
                xml="<svg width=\\"24\\" height=\\"19\\" viewBox=\\"0 0 24 19\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M1 4.64286L10.6433 11.03C11.045 11.2963 11.5171 11.4384 12 11.4384C12.4829 11.4384 12.955 11.2963 13.3567 11.03L23 4.64286M3.44444 18H20.5556C21.2039 18 21.8256 17.7441 22.284 17.2887C22.7425 16.8332 23 16.2155 23 15.5714V3.42857C23 2.78447 22.7425 2.16676 22.284 1.71131C21.8256 1.25587 21.2039 1 20.5556 1H3.44444C2.79614 1 2.17438 1.25587 1.71596 1.71131C1.25754 2.16676 1 2.78447 1 3.42857V15.5714C1 16.2155 1.25754 16.8332 1.71596 17.2887C2.17438 17.7441 2.79614 18 3.44444 18Z\\" stroke=\\"black\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"/>
</svg>
"
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M1 4.64286L10.6433 11.03C11.045 11.2963 11.5171 11.4384 12 11.4384C12.4829 11.4384 12.955 11.2963 13.3567 11.03L23 4.64286M3.44444 18H20.5556C21.2039 18 21.8256 17.7441 22.284 17.2887C22.7425 16.8332 23 16.2155 23 15.5714V3.42857C23 2.78447 22.7425 2.16676 22.284 1.71131C21.8256 1.25587 21.2039 1 20.5556 1H3.44444C2.79614 1 2.17438 1.25587 1.71596 1.71131C1.25754 2.16676 1 2.78447 1 3.42857V15.5714C1 16.2155 1.25754 16.8332 1.71596 17.2887C2.17438 17.7441 2.79614 18 3.44444 18Z"
                    propList={
                      Array [
                        "stroke",
                        "strokeWidth",
                        "strokeLinecap",
                        "strokeLinejoin",
                      ]
                    }
                    stroke={4278190080}
                    strokeLinecap={1}
                    strokeLinejoin={1}
                    strokeWidth={2}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <TouchableOpacity
              activeOpacity={0.2}
              style={
                Object {
                  "alignSelf": "flex-start",
                  "marginTop": 10,
                }
              }
              testID="clickable"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                    Object {
                      "color": "#FF7171",
                      "fontSize": 16,
                    },
                    Object {},
                  ]
                }
              >
                Need help finding your inmate ID?
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              style={
                Object {
                  "alignSelf": "flex-start",
                  "marginTop": 10,
                }
              }
              testID="clickable"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                    Object {
                      "color": "#FF7171",
                      "fontSize": 16,
                    },
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FF7171",
                    }
                  }
                >
                  Tap here to search the
                   
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Poppins-Bold",
                        },
                        Object {
                          "color": "#FF7171",
                        },
                      ]
                    }
                  >
                    CT
                  </Text>
                   
                  database
                  .
                </Text>
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              style={
                Object {
                  "alignSelf": "flex-start",
                  "marginBottom": 30,
                  "marginTop": 10,
                }
              }
              testID="clickable"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                    Object {
                      "color": "#FF7171",
                      "fontSize": 16,
                    },
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FF7171",
                    }
                  }
                >
                  Tap here to search the
                   
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Poppins-Bold",
                        },
                        Object {
                          "color": "#FF7171",
                        },
                      ]
                    }
                  >
                    Federal
                  </Text>
                   
                  database
                  .
                </Text>
              </Text>
            </TouchableOpacity>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 8,
                  "height": 60,
                  "marginBottom": 10,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
              testID="parent"
            >
              <View
                testID="invalid"
              />
              <View
                testID="unfocused"
              />
              <TextInput
                allowFontScaling={true}
                multiline={false}
                numberOfLines={1}
                placeholder="State"
                rejectResponderTermination={true}
                secureTextEntry={false}
                style={
                  Array [
                    Object {
                      "borderColor": "rgba(0,0,0,0)",
                      "borderRadius": 8,
                      "borderWidth": 2,
                      "color": "#9A9A9A",
                      "flex": 1,
                      "fontSize": 16,
                      "padding": 10,
                      "textAlignVertical": "top",
                      "width": "100%",
                    },
                    Object {},
                    Object {},
                    Object {
                      "height": 50,
                    },
                    Object {},
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <View />
              <View />
              <View
                style={
                  Object {
                    "height": 200,
                    "width": "100%",
                    "zIndex": 5,
                  }
                }
              >
                <View />
              </View>
            </View>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 8,
                  "marginBottom": 10,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
              testID="parent"
            >
              <View
                testID="invalid"
              />
              <View
                testID="unfocused"
              />
              <TextInput
                allowFontScaling={true}
                multiline={false}
                numberOfLines={1}
                placeholder="First Name"
                rejectResponderTermination={true}
                secureTextEntry={false}
                style={
                  Array [
                    Object {
                      "borderColor": "rgba(0,0,0,0)",
                      "borderRadius": 8,
                      "borderWidth": 2,
                      "color": "#9A9A9A",
                      "flex": 1,
                      "fontSize": 16,
                      "padding": 10,
                      "textAlignVertical": "top",
                      "width": "100%",
                    },
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <View />
              <View />
              <View
                style={
                  Object {
                    "height": 0,
                    "width": "100%",
                    "zIndex": 5,
                  }
                }
              >
                <View />
              </View>
            </View>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 8,
                  "marginBottom": 10,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
              testID="parent"
            >
              <View
                testID="invalid"
              />
              <View
                testID="unfocused"
              />
              <TextInput
                allowFontScaling={true}
                multiline={false}
                numberOfLines={1}
                placeholder="Last Name"
                rejectResponderTermination={true}
                secureTextEntry={false}
                style={
                  Array [
                    Object {
                      "borderColor": "rgba(0,0,0,0)",
                      "borderRadius": 8,
                      "borderWidth": 2,
                      "color": "#9A9A9A",
                      "flex": 1,
                      "fontSize": 16,
                      "padding": 10,
                      "textAlignVertical": "top",
                      "width": "100%",
                    },
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <View />
              <View />
              <View
                style={
                  Object {
                    "height": 0,
                    "width": "100%",
                    "zIndex": 5,
                  }
                }
              >
                <View />
              </View>
            </View>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 8,
                  "marginBottom": 10,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
              testID="parent"
            >
              <View
                testID="invalid"
              />
              <View
                testID="unfocused"
              />
              <TextInput
                allowFontScaling={true}
                multiline={false}
                numberOfLines={1}
                placeholder="Inmate Number"
                rejectResponderTermination={true}
                secureTextEntry={false}
                style={
                  Array [
                    Object {
                      "borderColor": "rgba(0,0,0,0)",
                      "borderRadius": 8,
                      "borderWidth": 2,
                      "color": "#9A9A9A",
                      "flex": 1,
                      "fontSize": 16,
                      "padding": 10,
                      "textAlignVertical": "top",
                      "width": "100%",
                    },
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <View />
              <View />
              <View
                style={
                  Object {
                    "height": 0,
                    "width": "100%",
                    "zIndex": 5,
                  }
                }
              >
                <View />
              </View>
            </View>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 8,
                  "height": 60,
                  "marginBottom": 0,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
              testID="parent"
            >
              <View
                testID="invalid"
              />
              <View
                testID="unfocused"
              />
              <TextInput
                allowFontScaling={true}
                multiline={false}
                numberOfLines={1}
                placeholder="Relationship to Inmate"
                rejectResponderTermination={true}
                secureTextEntry={false}
                style={
                  Array [
                    Object {
                      "borderColor": "rgba(0,0,0,0)",
                      "borderRadius": 8,
                      "borderWidth": 2,
                      "color": "#9A9A9A",
                      "flex": 1,
                      "fontSize": 16,
                      "padding": 10,
                      "textAlignVertical": "top",
                      "width": "100%",
                    },
                    Object {},
                    Object {},
                    Object {
                      "height": 50,
                    },
                    Object {},
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <View />
              <View />
              <View
                style={
                  Object {
                    "height": 200,
                    "width": "100%",
                    "zIndex": 5,
                  }
                }
              >
                <View />
              </View>
            </View>
          </View>
        </ScrollView>
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FFF5F5",
              "flexDirection": "row",
              "padding": 15,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={1}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FF7171",
                  "borderRadius": 16,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginVertical": 5,
                  "paddingHorizontal": 14,
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FF9E9E",
                  "borderRadius": 16,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginVertical": 5,
                },
                Object {
                  "flex": 1,
                  "margin": 10,
                },
                Object {},
                Object {
                  "elevation": 4,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.22,
                },
              ]
            }
            testID="clickable"
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Poppins-Medium",
                  },
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "700",
                    "padding": 6,
                  },
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "700",
                    "padding": 6,
                  },
                  Object {},
                  Object {},
                ]
              }
            >
              Next
            </Text>
            <View
              style={
                Object {
                  "position": "absolute",
                  "right": 12,
                }
              }
              testID="nextIcon"
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={21}
                bbWidth={24}
                fill="none"
                focusable={false}
                height={21}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 21,
                      "width": 24,
                    },
                  ]
                }
                vbHeight={21}
                vbWidth={24}
                width={24}
                xml="<svg width=\\"24\\" height=\\"21\\" viewBox=\\"0 0 24 21\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<path d=\\"M24 10.5L16.2772 21H12.1188L18.6535 12.25H0V8.75H18.6535L12.1188 0H16.2772L24 10.5Z\\" fill=\\"white\\"/>
</svg>
"
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M24 10.5L16.2772 21H12.1188L18.6535 12.25H0V8.75H18.6535L12.1188 0H16.2772L24 10.5Z"
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
